test/cri-containerd/container_test.go

```diff
@@ -131,3 +140,105 @@ func Test_RotateLogs_LCOW(t *testing.T) {
 		expected++
 	}
 }
+
+func Test_RunContainer_Events_LCOW(t *testing.T) {
+	pullRequiredLcowImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+	client := newTestRuntimeClient(t)
+
+	podctx, podcancel := context.WithCancel(context.Background())
+	defer podcancel()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	targetNamespace := "k8s.io"
+
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Uid:       "0",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	topicNames, filters := getTargetRunTopics()
+	eventService := newTestEventService(t)
+	stream, errs := eventService.Subscribe(ctx, filters...)
+
+	bufferedEvents := make([]*events.Envelope, len(topicNames))
+	var eventErrs []error
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	// buffer events here so we can check the ID later
+	go func() {
+		for i := 0; i < len(bufferedEvents); i++ {
+			select {
+			case env := <-stream:
+				bufferedEvents[i] = env
+			case e := <-errs:
+				eventErrs = append(eventErrs, errors.Errorf("event subscription err %v", e))
```

Comment:
We could just immediately `t.Fatal` here if we hit an error, rather than adding it to a slice.

Reply:
Since this is being run in a separate go routine, running t.Fatal here would not actually fail the test.